tests/drivers/sensor/accel/src/main.c

```diff
@@ -55,17 +55,33 @@ static void run_tests_on_accel(const char *label)
 	zassert_not_null(accel, "Unable to get Accelerometer device");
 	k_object_access_grant(accel, k_current_get());
 	accel_label = label;
-	ztest_test_suite(test_sensor_accel,
-			 ztest_user_unit_test(test_sensor_accel_basic));
-	ztest_run_test_suite(test_sensor_accel);
+	test_sensor_accel_basic();
 }
 
-/* test case main entry */
-void test_main(void)
+static void test_accel_0(void)
 {
 	run_tests_on_accel(DT_LABEL(DT_ALIAS(accel_0)));
+}
+
+static void test_accel_1(void)
+{
+	run_tests_on_accel(DT_PROP_OR(DT_ALIAS(accel_1), label, ""));
+}
+
+static bool test_accel_1_pragma(const void *state)
+{
+	ARG_UNUSED(state);
+
+	return DT_NODE_EXISTS(DT_ALIAS(accel_1));
+}
+
+ztest_register_test_suite(test_sensor_accel0, NULL,
+			  ztest_user_unit_test(test_accel_0));
+ztest_register_test_suite(test_sensor_accel1, test_accel_1_pragma,
+			  ztest_user_unit_test(test_accel_1));
 
-#if DT_NODE_EXISTS(DT_ALIAS(accel_1))
-	run_tests_on_accel(DT_LABEL(DT_ALIAS(accel_1)));
-#endif
```

Comment:
`DT_NODE_EXISTS(DT_ALIAS(accel_1))` is statically computable, so I am not sure what benefit this can offer over the good old `#if` approach -- if any, it makes it harder to understand the overall flow by moving the control statement to elsewhere.

This might make sense in the languages that support lambda expression and the pragma can be declared as a lambda, but that is unfortunately not the case in C -- even then, its usage should be limited to non-statically-computable cases.